nimbleparse/src/main.rs

```diff
@@ -346,24 +346,156 @@ fn main() {
+                            let mut paths = paths.peekable();
+                            if paths.peek().is_none() {
+                                eprintln!(
+                                    "test_files glob '{}' produced no input to test at path",
+                                    s
+                                );
+                                process::exit(1);
+                            }
+                            let mut ret: Vec<PathBuf> = Vec::new();
+                            for path in paths {
+                                match path {
+                                    Ok(path) => {
+                                        ret.push(path);
+                                    }
+                                    Err(e) => {
+                                        eprintln!(
+                                            "Error globbing %grmtools 'test_files' {}",
+                                            e.to_string()
+                                        );
+                                        process::exit(1);
+                                    }
+                                }
+                            }
+                            ret
+                        } else {
+                            eprintln!(
+                                "Error globbing %grmtools 'test_files', {}",
+                                paths.unwrap_err().to_string()
+                            );
+                            process::exit(1);
+                        }
+                    }
+                    Value::Setting(Setting::Num(_, loc)) => {
+                        eprintln!(
+                            "Expected 'test_files: \"glob\"' found bool in '%grmtools' from '{:?}'",
+                            loc
+                        );
+                        process::exit(1);
+                    }
+                    Value::Setting(Setting::Unitary(Namespaced {
+                        namespace: _,
+                        member: (member, memb_loc),
+                    })) => {
+                        eprintln!("Expected 'test_files: \"glob\"' found {member} in '%grmtools' from '{:?}'", memb_loc);
+                        process::exit(1);
+                    }
+                    Value::Setting(Setting::Constructor {
+                        ctor:
+                            Namespaced {
+                                namespace: _,
+                                member: (ctor_memb, ctor_memb_loc),
+                            },
+                        arg:
+                            Namespaced {
+                                namespace: _,
+                                member: (arg_memb, _),
+                            },
+                    }) => {
+                        eprintln!("Expected 'test_files: \"glob\"' found constuctor '{:?}::{:?}' in '%grmtools' from: {:?}", ctor_memb, arg_memb, ctor_memb_loc);
```

Comment:
So, the only thoughts I've had for cleaning this up is `TryFrom<Value<T>> for String`,
The only thing being that doesn't allow us to pass the `test_files` key, nor mention globs.

I suppose we could implement it for a tuple `TryFrom<(String, Value<T>)>` passing the key -- or even a partial error string

Reply:
In [621a0df](https://github.com/softdevteam/grmtools/pull/559/commits/621a0df307155fcd11b3b03cfabd0e2c12a50772) I didn't try and abuse `TryFrom` which was lacking context, or doing `TryFrom` for a tuple, which just seemed like a hack...

Instead I added a kind of ad-hoc method `expect_string_with_context`, It seems like it improved things only a little.  I don't have much in the way of ideas for cleaning it up further yet